src/_pytest/compat.py

```diff
@@ -313,16 +313,26 @@ def safe_isclass(obj: object) -> bool:
         return False
 
 
-def get_user_id() -> int | None:
-    """Return the current user id, or None if we cannot get it reliably on the current platform."""
-    # win32 does not have a getuid() function.
-    # On Emscripten, getuid() is a stub that always returns 0.
-    if sys.platform in ("win32", "emscripten"):
+def get_user_id(*, UNRELIABLE: int = -1) -> int | None:
```

Comment:
Did you mean this extra argument as some future-proofing here?

I think this is the wrong approach: the caller should not *need* to know which is the value that represents an error from `os.getuid`. If it ever varies between platforms, it should be handled internally by `get_user_id` itself.

Now, if your intent is to improve reliability, instead of the hardcoded value we had before, you can move it to a constant:

```python
UNRELIABLE = -1
return uid if uid != UNRELIABLE else None
```

(Also parameters should be camel case).